### PR TITLE
IGNITE-20340 Move handler of AlterTable command from CatalogManager to command itself

### PR DESCRIPTION
**Pull request by korlov42**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/commands/AlterTableDropColumnCommandValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests to verify validation of {@link AlterTableDropColumnCommand}.
+ */
+@SuppressWarnings({"ThrowableNotThrown"})
+public class AlterTableDropColumnCommandValidationTest extends AbstractCommandValidationTest {
```
Comment (AMashenkov) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/commands/AlterTableDropColumnCommandValidationTest.java:37]: I can't find a test, which checks that PK column can't be dropped.
Is it missed?

**Review by AMashenkov — APPROVED** [review]

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/commands/AlterTableDropColumnCommandValidationTest.java (hunk 1 above)
Comment (korlov42) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/commands/AlterTableDropColumnCommandValidationTest.java:37]: These tests resides in `CatalogManagerSelfTest` (see `testDropColumnWithIndexColumns` and `testDropColumnWithPrimaryKeyColumns`). But I agree it's better to have dedicated validation tests for this cases too.

Added required tests

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ddl/DdlCommandHandler.java [hunk 2]
```diff
@@ -338,6 +339,10 @@ protected static BiFunction<Object, Throwable, Boolean> handleModificationResult
                 }
             }
 
+            if (err instanceof CatalogValidationException) {
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ddl/DdlCommandHandler.java:342]: may be better place for the convertation at SqlExceptionMapperProvider

Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ddl/DdlCommandHandler.java]: moved to SqlExceptionMapperProvider

**Review by ygerzhedovich — APPROVED** [review]

**Review by AMashenkov — APPROVED** [review]

**Pull request merged**